frontends/main/src/page-components/LearningResourceExpanded/AiChatSyllabusSlideDown.tsx

```diff
@@ -113,60 +140,99 @@ const getInitialMessage = (
   ]
 }
 
+export const AiChatSyllabusOpener = ({
+  open,
+  className,
+  onToggleOpen,
+}: {
+  open: boolean
+  className?: string
+  onToggleOpen: (open: boolean) => void
+}) => {
+  return (
+    <Opener className={className}>
+      <StyledButton
+        variant="bordered"
+        edge="rounded"
+        aria-pressed={open}
+        open={open}
+        onClick={() => onToggleOpen(!open)}
+      >
+        <RiSparkling2Line />
+        <Typography variant="body1">
+          Ask<strong>TIM</strong> about this course
+        </Typography>
+        {open ? <CloseButton /> : <OpenChevron />}
+      </StyledButton>
+    </Opener>
+  )
+}
+
 const AiChatSyllabusSlideDown = ({
   resource,
-  onToggleOpen,
   open,
+  onTransitionEnd,
+  scrollElement,
+  contentTopPosition,
+  chatTransitionState,
 }: {
   resource?: LearningResource
   open: boolean
-  onToggleOpen: (open: boolean) => void
+  onTransitionEnd: () => void
+  scrollElement: HTMLElement | null
+  contentTopPosition: number
+  chatTransitionState: ChatTransitionState
 }) => {
   const user = useUserMe()
+  const ref = useRef<HTMLDivElement>(null)
+
+  useEffect(() => {
+    const element = ref.current
+    const _onTransitionEnd = (event: TransitionEvent) => {
+      if (event.target === element) {
```

Comment:
i guess you could also check propertyName https://developer.mozilla.org/en-US/docs/Web/API/TransitionEvent/propertyName